### PR TITLE
Set default datagen log level to INFO

### LINKED ISSUE #443: Set default loglevel to INFO
After https://github.com/borntyping/rust-simple_logger/pull/31 is checked in, we should update the default loglevel in the two binary tools (icu4x-cldr-export and icu4x-gen-testdata) to INFO so that users get more useful information by default.

### LINKED ISSUE #925: Upgrade to SimpleLogger 1.12.0
My feature request was accepted to SimpleLogger.  We can clean up the call site now in `icu4x-datagen`.  See https://github.com/borntyping/rust-simple_logger/pull/31#issuecomment-889045595

https://crates.io/crates/simple_logger/1.12.0

### PR DESCRIPTION
**Pull request by Manishearth**
Fixes #443, fixes #925

cc @sffc

### CODE REVIEW

**Review by echeran — APPROVED** [review]

**Review by sffc — COMMENTED** [review]
lgtm!

File: tools/datagen/src/bin/datagen.rs [hunk 1]
```diff
@@ -207,7 +207,11 @@ fn main() -> anyhow::Result<()> {
         .get_matches();
 
     match matches.occurrences_of("VERBOSE") {
-        0 => SimpleLogger::from_env().init().unwrap(),
+        0 => SimpleLogger::new()
+            .env()
+            .with_level(log::LevelFilter::Info)
+            .init()
+            .unwrap(),
         1 => SimpleLogger::new()
```
Comment (sffc) [inline tools/datagen/src/bin/datagen.rs:215]: Nit: Now that Info is the default, consider promoting Trace to a single v and drop the double vv

File: tools/datagen/Cargo.toml [hunk 2]
```diff
@@ -39,7 +39,7 @@ icu_testdata = { version = "0.3", path = "../../provider/testdata", features = [
 log = "0.4"
 reqwest = { version = "0.11", features = ["json", "stream", "gzip"] }
 serde = { version = "1.0", default-features = false, features = ["derive", "alloc"] }
-simple_logger = "1.11"
+simple_logger = "1.12"
```
Comment (sffc) [inline tools/datagen/Cargo.toml:42]: Nit: Also please look in the docs and Makefile.toml files and see if we can remove the -v from anywhere.

**Review by sffc — COMMENTED** [review]
Post-submit review comments

**Pull request merged**